static/app/views/alerts/rules/issue/details/ruleDetails.tsx

```diff
@@ -61,9 +63,47 @@ class AlertRuleDetails extends AsyncComponent<Props, State> {
     });
   }
 
-  componentDidUpdate(prevProps: Props) {
+  async componentDidUpdate(prevProps: Props) {
     const {organization: prevOrg, params: prevParams} = prevProps;
-    const {organization: currOrg, params: currParams} = this.props;
+    const {organization: currOrg, params: currParams, location} = this.props;
+
+    const {ruleId, projectId: projectSlug} = currParams;
+
+    const rule = this.state.rule;
+    const urlParams = new URLSearchParams(location.search);
+    if (urlParams.get('mute') === '1' && rule && !rule.snooze) {
+      try {
+        await this.api.requestPromise(
+          `/projects/${currOrg.slug}/${projectSlug}/rules/${ruleId}/snooze/`,
+          {
+            method: 'POST',
+            data: {
+              target: 'me',
+            },
+          }
+        );
+
+        urlParams.delete('mute');
+        browserHistory.push({
```

Comment:
I think a `browserHistory.replace` makes more sense here. You wouldn't want the back button to add back the mute param.